src/library_glemu.js

```diff
@@ -3415,9 +3652,99 @@ var LibraryGLEmulation = {
   glReadBuffer: function() { throw 'glReadBuffer: TODO' },
 #endif
 
-  glLightfv: function() { throw 'glLightfv: TODO' },
-  glLightModelfv: function() { throw 'glLightModelfv: TODO' },
-  glMaterialfv: function() { throw 'glMaterialfv: TODO' },
+  glClipPlane: function(pname, param) {
+    if ((pname >= 0x3000) && (pname < 0x3006)  /* GL_CLIP_PLANE0 to GL_CLIP_PLANE5 */) {
+      var clipPlaneId = pname - 0x3000;
+
+      GLEmulation.clipPlaneEquation[clipPlaneId][0] = {{{ makeGetValue('param', '0', 'double') }}};
+      GLEmulation.clipPlaneEquation[clipPlaneId][1] = {{{ makeGetValue('param', '8', 'double') }}};
+      GLEmulation.clipPlaneEquation[clipPlaneId][2] = {{{ makeGetValue('param', '16', 'double') }}};
+      GLEmulation.clipPlaneEquation[clipPlaneId][3] = {{{ makeGetValue('param', '24', 'double') }}};
+
+      // apply inverse transposed current modelview matrix when setting clip plane
+      var tmpMV = GLImmediate.matrixLib.mat4.create(GLImmediate.matrix[0]);
+      GLImmediate.matrixLib.mat4.inverse(tmpMV);
+      GLImmediate.matrixLib.mat4.transpose(tmpMV);
+      GLImmediate.matrixLib.mat4.multiplyVec4(tmpMV, GLEmulation.clipPlaneEquation[clipPlaneId]);
+    }
+  },
+
+  glLightfv: function(light, pname, param) {
+      if ((light >= 0x4000) && (light < 0x4008)  /* GL_LIGHT0 to GL_LIGHT7 */) {
```

Comment:
indentation here and in other places looks incorrect - it should always be 2 space (not 4, and not tabs).

Reply:
Sorry for that. Fixed it!